waddrmgr/address.go

```diff
@@ -225,7 +249,7 @@ func (a *managedAddress) ExportPubKey() string {
 // This is part of the ManagedPubKeyAddress interface implementation.
 func (a *managedAddress) PrivKey() (*btcec.PrivateKey, error) {
 	// No private keys are available for a watching-only address manager.
-	if a.manager.watchingOnly {
+	if a.manager.watchingOnly || a.watchingOnly {
```

Comment:
I suspect this should now just be `if a.watchingOnly {`.  All addreses under a watching-only address manager must now be marked watching-only too, so this could hide a bug by checking both conditions.